src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -85,11 +82,32 @@ public ProgrammingExercise importProgrammingExerciseFromFile(ProgrammingExercise
         if (Boolean.TRUE.equals(programmingExerciseForImport.isStaticCodeAnalysisEnabled())) {
             staticCodeAnalysisService.createDefaultCategories(importedProgrammingExercise);
         }
+        copyEmbeddedFiles(exerciseFilePath.toAbsolutePath().getParent().resolve(FileNameUtils.getBaseName(exerciseFilePath.toString())));
         importRepositoriesFromFile(importedProgrammingExercise, importExerciseDir, oldShortName);
         importedProgrammingExercise.setCourse(course);
+        fileService.scheduleForDirectoryDeletion(importExerciseDir, 5);
         return importedProgrammingExercise;
     }
 
+    /**
+     * Copy embedded files from the extracted zip file to the markdown folder, so they can be used in the problem statement
+     *
+     * @param importExerciseDir the directory where the extracted zip file is located
+     **/
+    private void copyEmbeddedFiles(Path importExerciseDir) throws IOException {
+        Path embeddedFilesDir = importExerciseDir.resolve("files");
+
+        if (!Files.exists(embeddedFilesDir)) {
+            return;
+        }
+        try (var embeddedFiles = Files.list(embeddedFilesDir)) {
+            for (var file : embeddedFiles.toList()) {
+                var targetPath = Path.of(FilePathService.getMarkdownFilePath(), file.getFileName().toString());
```

Comment:
Does this overwrite any other file with the same name that is already stored in Artemis?

Reply:
No, it does nothing if it is the same file or fails if the file already exists.
It's very unlikely that this file already exists and the method fails because the file name contains the current timestamp and a random UUID, so I thought this behavior is acceptable as long as it doesn't overwrite existing files.